flink-core/src/main/java/org/apache/flink/configuration/GlobalConfiguration.java

```diff
@@ -144,8 +144,15 @@ private static Configuration loadYAMLResource(File file) {
 						continue;
 					}
 
-					LOG.debug("Loading configuration property: {}, {}", key, value);
 					config.setString(key, value);
+
+					//to prevent logging the secure cookie
+					if(key.equals(ConfigConstants.SECURITY_COOKIE) && value != null) {
+						value = "******";
```

Comment:
Can all suppressed keys have a common prefix?

This should also be guarded by a test ;-)